src/daft-catalog/src/lib.rs

```diff
@@ -0,0 +1,136 @@
+mod data_catalog;
+mod data_catalog_table;
+pub mod errors;
+
+// Export public-facing traits
+use std::{collections::HashMap, default, sync::Arc};
+
+use daft_logical_plan::LogicalPlanBuilder;
+pub use data_catalog::DataCatalog;
+pub use data_catalog_table::DataCatalogTable;
+
+#[cfg(feature = "python")]
+pub mod python;
+
+use errors::Error;
+
+pub mod global_catalog {
+    use std::sync::{Arc, Mutex};
+
+    use lazy_static::lazy_static;
+
+    use crate::{DaftMetaCatalog, DataCatalog};
+
+    lazy_static! {
+        pub(crate) static ref GLOBAL_DAFT_META_CATALOG: Mutex<DaftMetaCatalog> =
+            Mutex::new(DaftMetaCatalog::new_from_env());
+    }
+
+    /// Register a DataCatalog with the global DaftMetaCatalog
+    pub fn register_catalog(catalog: Arc<dyn DataCatalog>, name: Option<&str>) {
+        GLOBAL_DAFT_META_CATALOG
+            .lock()
+            .unwrap()
+            .register_catalog(catalog, name);
+    }
+
+    /// Unregisters a catalog with the global DaftMetaCatalog
+    pub fn unregister_catalog(name: Option<&str>) -> bool {
+        GLOBAL_DAFT_META_CATALOG
+            .lock()
+            .unwrap()
+            .unregister_catalog(name)
+    }
+}
+
+/// Name of the default catalog
+static DEFAULT_CATALOG_NAME: &str = "default";
+
+/// The [`DaftMetaCatalog`] is a catalog of [`DataCatalog`] implementations
+///
+/// Users of Daft can register various [`DataCatalog`] with Daft, enabling
+/// discovery of tables across various [`DataCatalog`] implementations.
+pub struct DaftMetaCatalog {
```

Comment:
What do you mean by "persist"? Are you referring to:

1 Save tables to a catalog (this is planned for later work)
2 Load catalog from disk (this is also planned for later work, the idea is to load it from a YAML or something at `~/.daft.yaml`)
3 Save the current state of the DaftMetaCatalog to disk somehow

Unfortunately because we're using the Python catalog implementations and external libraries, I don't think there is an easy way to do (3) without relying on pickling of those Python libraries. I think the best we can do is read from some kind of configuration file (option 2).

Reply:
yeah i was thinking mostly of (3). I think it's fine if we can't support this now, but we should definitely keep it on our radar!